controlalgorithm/persistent_data.py

```diff
@@ -129,4 +129,31 @@ def get_cloud_cover_percentage_and_ext_temp():
     # otherwise cloud cover and ext temp values not in persistent_data
     else:
         cloud_cover_percentage, ext_temp_celsius = update_cloud_cover_percentage_and_ext_temp(lat, lon, timezone_adjustment)    
-    return cloud_cover_percentage, ext_temp_celsius
\ No newline at end of file
+    return cloud_cover_percentage, ext_temp_celsius
+
+"""
+Read and return the motor position (constrained from -90 to 90 in degrees)
+"""
+def get_motor_position():
+    with open("persistent_data.json", "r+") as fp:
+        persistent_data_dict = json.load(fp)
+
+    if "motor_position" not in persistent_data_dict:
+        persistent_data_dict["motor_position"] = 0
+    
+    return persistent_data_dict["motor_position"]
+    
+"""
+Update the motor position (constrained from -90 to 90 in degrees)
+"""
+def set_motor_position(angle):
+    with open("persistent_data.json", "r+") as fp:
```

Comment:
Do you have anything that tests this file i/o? I know when i used files for stuff, it would break depending on where the code was run from. Any way you could make this path relative to the file or something else? See the tests for blinds schedule for reference, I think I had to do it there.

Reply:
I've added a variable persistent_data_path to point to where the json file is written/read in persistent_data.py
Will use it after Ishaat tests other changes
Might just put it in a separate PR